File: go/invert_binary_tree.go
package main

import "fmt"
import "leet_code/tree"

func main()  {
	node := listNode(testCase1)
	ret := invertTree(node)
	fmt.Println(ret.Left.Val)
}

func invertTree(root *tree.TreeNode) *tree.TreeNode {
	if root == nil {
		return nil
	}
	right := invertTree(root.Right)
	left := invertTree(root.Left)
	root.Right = left
	root.Left = right
	return root
}

var testCase1 = []int{4,2,7,1,3,6,9}

// 平衡二叉树
func listNode(arr []int) *tree.TreeNode {
	l := len(arr)
	node := tree.TreeNode{Val: arr[0]}
	for i := 1;i < l;i++ {
		subNode := tree.TreeNode{}
		subNode.Val = i
		if i % 2 == 0 {

			node.Left = &subNode
		}else{
			node.Right = &subNode
		}
	}
	return &node
}

File: php/invert_binary_tree.php
<?php

// 226.Given the root of a binary tree, invert the tree, and return its root.
// 翻转一棵二叉树。
//Definition for a binary tree node.
class TreeNode {
    public $val = null;
    public $left = null;
    public $right = null;
    function __construct($val = 0, $left = null, $right = null) {
        $this->val = $val;
        $this->left = $left;
        $this->right = $right;
    }
}

class Solution {

    /**
     * @param TreeNode $root
     * @return TreeNode
     */
    function invertTree($root) {
        if($root == null){
            return null;
        }
        $right = $this->invertTree($root->right);
        $left = $this->invertTree($root->left);
        $root->left = $right;
        $root->right = $left;
        return $root;
    }
}